js/renovation/ui/editors/check_box/check_box.tsx

```diff
@@ -139,12 +142,18 @@ export class CheckBox extends JSXComponent<CheckBoxPropsType>() {
   }
 
   onWidgetClick(event: Event): void {
-    const { readOnly, saveValueChangeEvent } = this.props;
-    const value = this.props.value ?? false;
+    const {
+      value, readOnly, allowIndeterminateStateByClick, saveValueChangeEvent,
+    } = this.props;
 
     if (!readOnly) {
       saveValueChangeEvent?.(event);
-      this.props.value = !value;
+
+      if (allowIndeterminateStateByClick) {
+        this.props.value = !isDefined(value) || (!value ? null : false);
```

Comment:
`isDefined` is an outdated api a little bit. We almost don't use it after "nullish coalescing" operator support. It's ok to use it if it's really necessary, but here we can just write `value === null` because value cannot be `undefined` here.

I think we can write it like
```suggestion
        this.props.value = value === null || (value === false ? null : false);
```

Reply:
https://github.com/DevExpress/DevExtreme/pull/22616/commits/97a7eeaea09ac17681847759a4663e6facf3e578